microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/MPOpenAPIBuilder.java.save

```diff
@@ -0,0 +1,413 @@
+/*
+ * Copyright (c) 2019, 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.openapi;
+
+import java.lang.System.Logger.Level;
+import java.lang.reflect.Modifier;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.helidon.config.metadata.Configured;
+import io.helidon.config.metadata.ConfiguredOption;
+import io.helidon.microprofile.server.JaxRsApplication;
+import io.helidon.microprofile.server.JaxRsCdiExtension;
+
+import io.smallrye.openapi.api.OpenApiConfig;
+import io.smallrye.openapi.api.OpenApiConfigImpl;
+import io.smallrye.openapi.runtime.scanner.FilteredIndexView;
+import jakarta.enterprise.inject.spi.CDI;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Application;
+import jakarta.ws.rs.core.Feature;
+import jakarta.ws.rs.ext.Provider;
+import org.eclipse.microprofile.config.Config;
+import io.smallrye.jandex.AnnotationInstance;
+import io.smallrye.jandex.AnnotationTarget;
+import io.smallrye.jandex.ClassInfo;
+import io.smallrye.jandex.DotName;
+import io.smallrye.jandex.IndexView;
+
+/**
+ * Fluent builder for OpenAPISupport in Helidon MP.
+ */
+@Configured(prefix = MPOpenAPIBuilder.MP_OPENAPI_CONFIG_PREFIX)
```

Comment:
we should delete this file, or move it under tests or something.

Reply:
For sure. This slipped through.